Update git branch protection docs (#8275)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/git/version-control-basics.md

| Branch | A branch is a parallel version of a repository. It is contained within the repository but does not affect the primary or main branch allowing you to work freely without disrupting the live version. When you've made the changes you want to make, you can merge your branch back into the main branch to publish your changes |
| Checkout | The `checkout` command is used to create a new branch, change your current working branch to a different branch, or switch to a different version of a file from a different branch. |
| Commit | A commit is a user’s change to a file (or set of files). When you make a commit to save your work, <Constant name="git" /> creates a unique ID that allows you to keep a record of the specific changes committed along with who made them and when. Commits usually contain a commit message which is a brief description of what changes were made. |

> | main | The primary, base branch of all repositories. All committed and accepted changes should be on the main branch.<br /><br /> In the <Constant name="cloud_ide" />, the main branch is protected. This means you can't directly edit, format, or lint files and execute dbt commands in your protected primary git branch. Keep in mind that all IDE Git activity is subject to the permissions of the user's configured credentials, and the rules configured at the remote (e.g. branch protection). Since the <Constant name="cloud_ide" /> prevents commits to the protected branch, you can commit those changes to a new branch.|

Author: ```suggestion
| main | The primary, base branch of all repositories. All committed and accepted changes should be on the main branch.<br /><br /> In the <Constant name="cloud_ide" />, the main branch is protected. This means you can't directly edit, format, or lint files and execute dbt commands in your protected primary git branch. Keep in mind that all IDE Git activity is subject to the permissions of your configured credentials, and the rules configured at the remote (for example, branch protection). Since the <Constant name="cloud_ide" /> prevents commits to the protected branch, you can commit those changes to a new branch.|
```

> | main | The primary, base branch of all repositories. All committed and accepted changes should be on the main branch.<br /><br /> In the <Constant name="cloud_ide" />, the main branch is protected. This means you can't directly edit, format, or lint files and execute dbt commands in your protected primary git branch. Keep in mind that all IDE Git activity is subject to the permissions of your configured credentials, and the rules configured at the remote (for example, branch protection). Since the <Constant name="cloud_ide" /> prevents commits to the protected branch, you can commit those changes to a new branch.|

Author: ```suggestion
| main | The primary, base branch of all repositories. All committed and accepted changes should be on the main branch.<br /><br /> In the <Constant name="cloud_ide" />, the main branch is protected. This means you can't directly edit, format, or lint files and execute dbt commands in your protected primary git branch. Keep in mind that all <Constant name="cloud_ide" /> Git activity is subject to the permissions of your configured credentials, and the rules configured at the remote (for example, branch protection). Since the <Constant name="cloud_ide" /> prevents commits to the protected branch, you can commit those changes to a new branch.|
```

> | main | The primary, base branch of all repositories. All committed and accepted changes should be on the main branch.<br /><br /> In the <Constant name="cloud_ide" />, the main branch is protected. This means you can't directly edit, format, or lint files, and execute dbt commands in your protected primary git branch. Keep in mind that all <Constant name="cloud_ide" /> Git activity is subject to the permissions of your configured credentials, and the rules configured at the repo (for example, branch protection). Since the <Constant name="cloud_ide" /> prevents commits to the protected branch, you can commit those changes to a new branch.|

Author: ```suggestion
| main | The primary, base branch of all repositories. All committed and accepted changes should be on the main branch.<br /><br /> In the <Constant name="cloud_ide" />, the main branch is protected. This means you can't directly edit, format, or lint files, and execute dbt commands in your protected primary git branch. Keep in mind that all <Constant name="cloud_ide" /> Git activity is subject to the permissions of your configured credentials, and the rules configured at the remote Git provider (for example, GitHub or GitLab branch protection). Since the <Constant name="cloud_ide" /> prevents commits to the protected branch, you can commit those changes to a new branch.|
```

> | main | The primary, base branch of all repositories. All committed and accepted changes should be on the main branch.<br /><br /> In the <Constant name="cloud_ide" />, the main branch is protected. This means you can't directly edit, format, or lint files and execute dbt commands in your protected primary git branch. Keep in mind that all <Constant name="cloud_ide" /> Git activity is subject to the permissions of your configured credentials, and the rules configured at the remote (for example, branch protection). Since the <Constant name="cloud_ide" /> prevents commits to the protected branch, you can commit those changes to a new branch.|

Reviewer: Is this supposed to be repo?
```suggestion
| main | The primary, base branch of all repositories. All committed and accepted changes should be on the main branch.<br /><br /> In the <Constant name="cloud_ide" />, the main branch is protected. This means you can't directly edit, format, or lint files, and execute dbt commands in your protected primary git branch. Keep in mind that all <Constant name="cloud_ide" /> Git activity is subject to the permissions of your configured credentials, and the rules configured at the repo (for example, branch protection). Since the <Constant name="cloud_ide" /> prevents commits to the protected branch, you can commit those changes to a new branch.|
```

> | main | The primary, base branch of all repositories. All committed and accepted changes should be on the main branch.<br /><br /> In the <Constant name="cloud_ide" />, the main branch is protected. This means you can't directly edit, format, or lint files, and execute dbt commands in your protected primary git branch. Keep in mind that all <Constant name="cloud_ide" /> Git activity is subject to the permissions of your configured credentials, and the rules configured at the remote Git provider (for example, GitHub or GitLab branch protection). Since the <Constant name="cloud_ide" /> prevents commits to the protected branch, you can commit those changes to a new branch.|

Reviewer 2: ```suggestion
| main | The primary, base branch of all repositories. All committed and accepted changes should be on the main branch.<br /><br /> In the <Constant name="cloud_ide" />, the main branch is protected. This means you can't directly edit, format, or lint files, and execute dbt commands in your protected primary git branch when using the <Constant name="dbt_platform" /> user interface. Keep in mind that all <Constant name="cloud_ide" /> Git activity is subject to the permissions of your configured credentials, and the rules configured at the remote Git provider (for example, GitHub or GitLab branch protection). Since the <Constant name="cloud_ide" /> prevents commits to the protected branch, you can commit those changes to a new branch.|
```

Reviewer 2: Sorry, I forgot to include this in the original message. Adding this bit makes it clear that the current restriction is also enforced on the UI currently, and this directly addresses the security angle that prompted this update. What do you think?

Author: @jp-dbt Great point! I updated it slightly to spell out UI and use "when using" instead of 'via". Just for clarity.
